src/pudl/transform/ferc714.py

```diff
@@ -315,35 +377,235 @@ def _post_process(df: pd.DataFrame, table_name: str) -> pd.DataFrame:
+            )
+        return df.drop(columns=["eia_code_count"])
+
+    @staticmethod
+    def clean_eia_codes_xbrl(xbrl: pd.DataFrame) -> pd.DataFrame:
+        """Make eia_code's cleaner coming from the XBRL data.
+
+        Desired outcomes here include all respondents have only one non-null
+        eia_code and all eia_codes that are actually the respondent_id_ferc714_xbrl
+        are nulled.
+        """
+        # we expect all of these submissions to be from the last Q
+        assert all(xbrl.report_period == "Q4")
+        # first we are gonna null out all of the "EIA" codes that are really just the respondent id
+        code_is_respondent_id_mask = xbrl.eia_code.str.startswith("C") & (
+            xbrl.respondent_id_ferc714_xbrl == xbrl.eia_code
+        )
+        xbrl.loc[code_is_respondent_id_mask, "eia_code"] = pd.NA
+
+        # lets null out some of the eia_code's from XBRL that we've manually culled
+        # because they are were determined to be wrong. These respondents
+        # had more than one value for their eia_code and one was always wrong
+        respondent_id_xbrl_to_bad_eia_code = {
+            "C002422": ["5776"],
+            "C011374": ["8376"],
+            "C002869": ["F720204"],
+            "C002732": ["F720204", "57049, 57050"],
+            "C011420": ["16606"],
+        }
+        for rid_xbrl, bad_eia_codes in respondent_id_xbrl_to_bad_eia_code.items():
+            xbrl.loc[
+                (xbrl.respondent_id_ferc714_xbrl == rid_xbrl)
+                & (xbrl.eia_code.isin(bad_eia_codes)),
+                "eia_code",
+            ] = pd.NA
+        return xbrl
+
+    @staticmethod
+    def convert_into_static_table_xbrl(xbrl: pd.DataFrame) -> pd.DataFrame:
+        """Convert this annually reported table into a skinnier, static table.
+
+        The CSV table is entirely static - it doesn't have any reported
+        changes that vary over time. The XBRL table does have start and end
+        dates in it. In order to merge these two sources, we are checking
+        whether or not the shared variables change over time and then
+        converting this table into a non-time-varying table.
+        """
+        # the CSV data does not vary by year, so we need to check if that is
+        # also going to be the case for the XBRL data. we check the eia_codes
+        # during ensure_eia_code_uniqueness. The name is less crucial but we
+        # should still check.
+        assert all(
+            xbrl.groupby(["respondent_id_ferc714_xbrl"])[  # noqa: PD101
+                ["respondent_name_ferc714"]
+            ].nunique()
+            == 1
+        )
+        cols_to_keep = [
+            "respondent_id_ferc714",
+            "respondent_id_ferc714_xbrl",
+            "respondent_name_ferc714",
+            "eia_code",
+        ]
+        # we are not checking whether the respondent_name_ferc714 is exactly
+        # consistent across the submissions so before we grab the one true eia_code
```

Comment:
Is this not what's happening in the assertion statement above?

Reply:
oh yea you are right